### PR TITLE
LPS-186452 | Test Automation | Master

### PR DESCRIPTION
**Pull request by victorhcunha**
[LPS-186468](https://liferay.atlassian.net/browse/LPS-186468)
[LPS-186467](https://liferay.atlassian.net/browse/LPS-186467)
[LPS-186452](https://liferay.atlassian.net/browse/LPS-186452)

### CODE REVIEW

**Comment by victorhcunha** [comment]
ci:test:sf

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectPersonalData.testcase [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+		JSONUser.tearDownNonAdminUsers();
+
+		var testPortalInstance = PropsUtil.get("test.portal.instance");
+
+		if (${testPortalInstance} == "true") {
+			PortalInstances.tearDownCP();
+		}
+	}
+
+	@description = "LPS-186452 - Verify that is possible to export personal data"
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectPersonalData.testcase:289]: `Verify a user's personal data can be exported`
I didn't correct this earlier, that's my mistake. 

Also, the ticket ID should be the story ticket and not the automation ticket

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectPersonalData.testcase [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+			ObjectAdmin.goToCustomObject(objectName = "CustomObject186452");
+		}
+
+		task ("And given several object entries added by the user") {
+			for (var entryNumber : list "1,2,3") {
+				ObjectAdmin.goToCustomObject(objectName = "CustomObject186452");
+
+				ObjectPortlet.addSingleFieldEntryViaUI(entry = "Entry ${entryNumber}");
+			}
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectPersonalData.testcase:339]: Why not add the entries using the API?

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectPersonalData.testcase [hunk 3]
```diff
@@ -22,6 +22,273 @@ definition {
 		}
 	}
 
+	@description = "LPS-186468 - Verify that is possible to anonymize object entries"
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectPersonalData.testcase:25]: `Verify the object entries can be anonymized`

Same thing here, the ticket ID should be the story ticket.

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectPersonalData.testcase [hunk 4]
```diff
@@ -22,6 +22,273 @@ definition {
...
+		}
+
+		task ("And given several object entries added by the user") {
+			ObjectAdmin.goToCustomObject(objectName = "CustomObject186468");
+
+			for (var entryName : list "B,C,D,E") {
+				ObjectAdmin.goToCustomObject(objectName = "CustomObject186468");
+
+				ObjectPortlet.addSingleFieldEntryViaUI(entry = "Entry ${entryName}");
+			}
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectPersonalData.testcase:80]: Same thing here, we should add the entries using the API instead

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectPersonalData.testcase [hunk 5]
```diff
@@ -22,6 +22,273 @@ definition {
...
+		task ("Then the author of the entries created by the user is Anonymous Anonymous") {
+			User.openUsersAdmin();
+
+			User.activateCP(
+				userName = "userfn userln",
+				userScreenName = "usersn");
+
+			User.logoutAndLoginPG(
+				userLoginEmailAddress = "userea@liferay.com",
+				userLoginFullName = "userfn userln");
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectPersonalData.testcase:112]: Do we need to login as the 2nd user here?

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectPersonalData.testcase [hunk 6]
```diff
@@ -22,6 +22,273 @@ definition {
...
+		}
+	}
+
+<<<<<<< HEAD
+=======
+	@description = "LPS-186467 - Verify that the entries of a object can be deleted"
+	@priority = 5
+	test CanDeleteObjectEntries {
+		property portal.acceptance = "true";
+
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectPersonalData.testcase]: Be careful when fixing SF errors or rebasing, this shouldn't be part of this commit. Please make sure this test is part of your third commit.

Also, don't forget to change the ticket ID to the story ticket.

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectPersonalData.testcase [hunk 7]
```diff
@@ -22,6 +22,273 @@ definition {
...
+			ObjectPortlet.selectCustomObject(label = "Custom Object 186467");
+
+			CreateObject.selectTitleField(fieldLabel = "Custom Field");
+
+			CreateObject.saveObject();
+
+			ObjectAdmin.openObjectAdmin();
+
+			ObjectPortlet.selectCustomObject(label = "Custom Object 186467");
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectPersonalData.testcase:169]: Do we need to navigate back to the object admin page here? Why not go straight to the layouts tab?

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectPersonalData.testcase [hunk 8]
```diff
@@ -22,6 +22,273 @@ definition {
...
+				userLoginEmailAddress = "userea@liferay.com",
+				userLoginFullName = "userfn userln");
+		}
+
+		task ("And given four object entries added by the user, Entry B, Entry C, Entry D and Entry E") {
+			for (var entryName : list "B,C,D,E") {
+				ObjectAdmin.goToCustomObject(objectName = "CustomObject186467");
+
+				ObjectPortlet.addSingleFieldEntryViaUI(entry = "Entry ${entryName}");
+			}
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectPersonalData.testcase:225]: Can we add the entries using the API instead?

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectPersonalData.testcase [hunk 9]
```diff
@@ -22,6 +22,273 @@ definition {
...
+
+			User.openUsersAdmin();
+
+			User.activateCP(
+				userName = "userfn userln",
+				userScreenName = "usersn");
+
+			User.logoutAndLoginPG(
+				userLoginEmailAddress = "userea@liferay.com",
+				userLoginFullName = "userfn userln");
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectPersonalData.testcase:277]: Do we need to login as the 2nd user here?

**Review by timpak — CHANGES_REQUESTED** [review]
Hey @victorhcunha please take a look at my commits below. Be careful when rebasing or fixing SF errors, you should make sure your changes are part of the right commit before forwarding it.

**Comment by victorhcunha** [comment]
ci:test:sf

**Comment by victorhcunha** [comment]
SF errors corrected, description tests updated (in the ticket too)
The next step is to refactor to do new user entries via API and remove unnecessary steps

**Comment by victorhcunha** [comment]
ci:test:sf

**Comment by victorhcunha** [comment]
Hi, @timpak !
The corrections were made.
Now all the entries are made via API, for that the logins became unnecessary and were removed. However, as the new user has no data on the portal other than the entries themselves, a new confirmation message is displayed when anonymizing and deleting the data, making the standard macros for this no longer suitable.

**Review by timpak — APPROVED** [review]

**Comment by timpak** [comment]
ci:forward

**Pull request closed without merging**